control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2294,6 +2284,33 @@ func (r *HostedControlPlaneReconciler) reconcileIgnition(ctx context.Context, hc
 		return fmt.Errorf("failed to reconcile api server ha proxy ignition config: %w", err)
 	}
 
+	imageContentSourceIgnitionConfig := manifests.ImageContentSourcePolicyIgnitionConfig(hcp.GetNamespace())
+	if !p.HasImageContentSourcePolicy {
+		// ensure the icsp configmap has been removed if no longer needed
+		err := r.Get(ctx, client.ObjectKeyFromObject(imageContentSourceIgnitionConfig), imageContentSourceIgnitionConfig)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed to check whether image content source policy configuration configmap exists")
```

Comment:
Please include the original error in the returned error, same issue below if the Delete call fails

Reply:
thanks, fixed